hawc/apps/animal/exports.py

```diff
@@ -1,744 +1,1520 @@
+            ):
+                (
+                    row["endpoint_group-lower_ci"],
+                    row["endpoint_group-upper_ci"],
+                ) = cont_ci(stdev, n, response)
+            elif (
+                data_type in [constants.DataType.DICHOTOMOUS, constants.DataType.DICHOTOMOUS_CANCER]
+                and incidence is not None
+            ):
+                (
+                    row["endpoint_group-lower_ci"],
+                    row["endpoint_group-upper_ci"],
+                ) = dich_ci(incidence, n)
+            return row
+
+        return df.apply(_func, axis="columns").drop(columns=["endpoint_group-stdev"])
+
+    def build_df(self) -> pd.DataFrame:
+        df = AnimalExporter().get_df(
+            self.queryset.select_related(
+                "animal_group__experiment__study",
+                "animal_group__dosing_regime",
+            )
+            .prefetch_related("groups", "animal_group__dosing_regime__doses")
+            .order_by("id", "groups", "animal_group__dosing_regime__doses")
+        )
+        df = df[
+            pd.isna(df["dose_group-id"])
+            | (df["endpoint_group-dose_group_id"] == df["dose_group-dose_group_id"])
+        ]
+        if df.empty:
+            return df
+        if obj := self.queryset.first():
+            doses = DoseUnits.objects.get_animal_units_names(obj.assessment_id)
+
+            df = df.assign(**{f"doses-{d}": None for d in doses})
+            df = self.handle_doses(df)
+        df["dosing_regime-dosed_animals"] = df["dosing_regime-dosed_animals"].astype(str)
+        df = self.handle_stdev(df)
+        df = self.handle_ci(df)
+
+        df = df.rename(
+            columns={
+                "endpoint_group-treatment_effect_display": "endpoint_group-treatment_effect",
+                "endpoint-expected_adversity_direction_display": "endpoint-expected_adversity_direction",
+                "experiment-type_display": "experiment-type",
+                "animal_group-sex_display": "animal_group-sex",
+                "dosing_regime-positive_control_display": "dosing_regime-positive_control",
+                "endpoint-effects_display": "endpoint-effects",
+                "animal_group-parents_display": "animal_group-parents",
+                "dosing_regime-route_of_exposure_display": "dosing_regime-route_of_exposure",
+                "endpoint-monotonicity_display": "endpoint-monotonicity",
+                "dosing_regime-negative_control_display": "dosing_regime-negative_control",
+                "endpoint-observation_time_units_display": "endpoint-observation_time_units",
+                "endpoint-trend_result_display": "endpoint-trend_result",
+                "endpoint-variance_type_name": "endpoint-variance_type",
+                "animal_group-species_name": "species-name",
+                "animal_group-strain_name": "strain-name",
+            }
+        )
+
+        return df
+
+
+class AnimalExporter2(Exporter):
```

Comment:
At the risk of making the `FlatFileExporter` cluttered, maybe these should be inner classes on those? ie:
```
class EndpointGroupFlatDataPivot(FlatFileExporter):
    _Exporter(Exporter):
        ...
```
That way we won't have to have nebulous naming on these, cause I'm not sure `EndpointGroupFlatDataPivotExporter` is any better of a name than `AnimalExporter2`.

Reply:
I agree; I don't like the AnimalExporter2,AnimalExporter3,AnimalExporter4 names. I think I'd prefer you second convention, which just has long names, but that's ok.

Maybe we can call it `EndpointGroupDataPivotExporter`?

Reply:
The convention I used is `{ClassName}Exporter` for the corresponding exporters.